File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
  const rank_lst = {}
  players.forEach((item, index) => 
  rank_lst[item] = index
  ) 
  for ( let call of callings ) {
    let call_idx = rank_lst[call]
    let front_name = players[call_idx-1]
    rank_lst[call] -= 1
    rank_lst[front_name] += 1
    players[call_idx-1] = players[call_idx]
    players[call_idx] = front_name
  }
  
  return players
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    # 각 player의 인덱스를 저장할 dictionary
    player_index = {player: idx for idx, player in enumerate(players)}
    
    for call in callings:
        idx = player_index[call]
        if idx > 0:
            # Swap players
            players[idx], players[idx-1] = players[idx-1], players[idx]
            # Update indices in the dictionary
            player_index[players[idx]] = idx
            player_index[players[idx-1]] = idx - 1
    
    return players